File: Sorting/MissingNumber.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
  sort(nums);

  for(let i = 0; i < nums.length; i++){
    if(nums[i] !== i) return i;
  }

  return nums.length;
};

function sort(nums) {
 let i = 0;

 while(i < nums.length){
   let currNum = nums[i];

   if(currNum < nums.length && currNum !== i){
     let temp = nums[currNum];
     nums[currNum] = nums[i];
     nums[i] = temp;
   } else {
     i++; 
   }
 }
}

File: Sorting/MissingNumber.cs
public class Solution {
    public int MissingNumber(int[] nums) {
      sort(nums);

      for(int i = 0; i < nums.Length; i++)
      {
        if(nums[i] != i)
          return i;
      }

      return nums.Length;
    }

    public static void sort(int[] nums)
    {
      int i = 0;
      while(i < nums.Length) 
      {
        int currInt = nums[i];
        
        if(currInt < nums.Length && currInt != nums[currInt])
        {
          int temp = nums[currInt];
          nums[currInt] = currInt;
          nums[i] = temp;
        }
        else 
        {
          i++;
        }
      }
    }
}